cupy/_core/include/cupy/carray.cuh

```diff
@@ -326,6 +326,16 @@ public:
     return strides_;
   }
 
+#ifdef CUPY_JIT_MODE
+  __device__ typename cupy::as_tuple<_ndim, ptrdiff_t>::type get_shape() const {
```

Comment:
Note: The `as_tuple` implementation by @asi1024 is extremely clever, but it wouldn't be needed if we move JIT from C++11 to C++14, where `std::index_sequence` (and its relatives) and `auto` as the return type  would just work...😉

Reply:
I am concerned that C++14 will work with old CUDA/HIP versions. I will check if CI passes with c++14 option in another PR.

Reply:
Thanks, @asi1024. I think C++11 is ok for now, we don't have to bump it right now, but it'd be good to explore a timeline for it 🙂